tests/common/fixtures/consistency_checker/consistency_checker.py

```diff
@@ -0,0 +1,335 @@
+        """
+        Queries the ASIC for the attributes of the objects in db_attributes which are the results
+        from the ASIC DB query.
+
+        Example return value:
+            {
+                "ASIC_STATE:SAI_OBJECT_TYPE_BUFFER_POOL:oid:0x18000000000628": {
+                    "SAI_BUFFER_POOL_ATTR_THRESHOLD_MODE": {
+                        "asicValue": "SAI_BUFFER_POOL_THRESHOLD_MODE_STATIC",
+                        "success": true
+                    },
+                    "SAI_BUFFER_POOL_ATTR_SIZE": {
+                        "success" false,
+                        "error": "Failed to query attribute value"
+                    },
+                    "SAI_BUFFER_POOL_ATTR_TYPE": {
+                        "asicValue": "SAI_BUFFER_POOL_TYPE_EGRESS",
+                        "success": true
+                    }
+                },
+                ...
+            }
+        """
+        # Map to format expected by the query-asic.py
+        asic_query = {k: list(v["value"].keys()) for k, v in db_attributes.items()}
+        asic_query_input_filename = f"query-input-{datetime.datetime.now().strftime('%Y%m%d%H%M%S')}.json"
+        with open(f"/tmp/{asic_query_input_filename}", 'w') as f:
+            json.dump(asic_query, f)
+
+        # Copy the input file to the syncd container
+        self._duthost.copy(src=f"/tmp/{asic_query_input_filename}", dest=f"/tmp/{asic_query_input_filename}")
+        self._duthost.shell((f"docker cp /tmp/{asic_query_input_filename} "
+                             f"{SYNCD_CONTAINER}:{DUT_DST_PATH_CONTAINER} && "
+                             f"rm /tmp/{asic_query_input_filename}"))
+
+        ld_lib_path_arg = "LD_LIBRARY_PATH=libsairedis-temp/usr/lib/x86_64-linux-gnu"\
+                          if self._libsairedis_download_url is not None else ""
+
+        res = self._duthost.shell((f"docker exec {SYNCD_CONTAINER} bash -c "
+                                   f"'cd {DUT_DST_PATH_CONTAINER} && "
+                                   f"{ld_lib_path_arg} python3 {DUT_SCRIPT_PATH_DST_CONTAINER} "
+                                   f"--input {asic_query_input_filename}'"))
+        if res['rc'] != 0:
+            raise Exception((f"Failed to query ASIC attributes. Return code: {res['rc']}, stdout: {res['stdout']}, "
+                             f"stderr: {res['stderr']}"))
+        asic_results = json.loads(res['stdout'])
+
+        return asic_results
+
+
+class ConsistencyCheckerProvider:
+    def is_consistency_check_supported(self, dut) -> bool:
+        """
+        Checks if the provided DUT is supported for consistency checking.
+        Currently, only supports select Arista platforms and SONiC versions 202305 and 202311.
+
+        :param dut: SonicHost object
+        :return bool: True if the DUT is supported, False otherwise
+        """
+
+        platform = dut.facts['platform']
+        if platform not in ["x86_64-arista_7060_cx32s", "x86_64-arista_7260cx3_64"]:
```

Comment:
You can move these list of supported platforms and OS versions to a constants file.

That way to add a new platform or OS support, core logic file remains untouched and constants file can grow independently.